build/gen_stub.php

```diff
@@ -3611,7 +3729,14 @@ protected function pName_FullyQualified(Name\FullyQualified $node) {
                 $fileInfo->generateFunctionEntries = true;
                 $fileInfo->declarationPrefix = $tag->value ? $tag->value . " " : "";
             } else if ($tag->name === 'generate-legacy-arginfo') {
-                $fileInfo->generateLegacyArginfo = true;
+                if ($tag->value && !in_array((int) $tag->value, ALL_PHP_VERSION_IDS, true)) {
+                    throw new Exception(
+                        "Legacy PHP version must be one of: \"" . PHP_70_VERSION_ID . "\" (PHP 7.0), \"" . PHP_80_VERSION_ID . "\" (PHP 8.0), " .
+                        "\"" . PHP_81_VERSION_ID . "\" (PHP 8.1), \"" . PHP_82_VERSION_ID . "\" (PHP 8.2), \"" . $tag->value . "\" provided"
+                    );
+                }
```

Comment:
Does it make sense to restrict here, or why not just say must be `>= 70000 && <= CURRENT_VERSION_ID` and it'll pick the nearest version. Like, I'm targeting PHP 7.3+. I just want to be able to write 70300, and be future "proof" against having php 7.0 specific things in my code.

Reply:
Yeah, you are right! The reason why I didn't add support for 7.x versions is because I forgot that they also have features (e.g. `void` return type) which need special handling. I can add support for these after feature freeze starts, because I'd like to work on something else before it.

Reply:
Another thing to consider is how the legacy PHP version should be provided in the stubs? Would it be better to use the exact version rather than the version ID?